Content.Server/MagicMirror/MagicMirrorSystem.cs

```diff
@@ -171,9 +256,9 @@ private void UpdateInterface(EntityUid uid, EntityUid playerUid, ICommonSession
 
     private void AfterUIOpen(EntityUid uid, MagicMirrorComponent component, AfterActivatableUIOpenEvent args)
     {
-        var looks = Comp<HumanoidAppearanceComponent>(args.User);
-        var actor = Comp<ActorComponent>(args.User);
+        if (!TryComp<HumanoidAppearanceComponent>(args.User, out var humanoid)) return;
+        component.Target = new Entity<HumanoidAppearanceComponent>(args.User, humanoid);
 
```

Comment:
Please don't inline like this it makes it unnecessarily hard to read, also don't store components on systems + this will break with map saving.